pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -500,35 +517,31 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 	}
 
 	// Iterate over members in each pool that _do_ exist
-	var poolID string
 	err = v2_pools.List(lbaas.network, v2_pools.ListOpts{LoadbalancerID: loadbalancer.ID}).EachPage(func(page pagination.Page) (bool, error) {
-		poolsList, err := v2_pools.ExtractPools(page)
+		poolsList, err = v2_pools.ExtractPools(page)
```

Comment:
ditto, after this loop `poolsList` will only contain the pools on the last page.